share/sysbuild/CMakeLists.txt

```diff
@@ -69,8 +69,17 @@ add_subdirectory(bootloader)
 # This allows for board and app specific images to be included.
 include(${BOARD_DIR}/sysbuild.cmake OPTIONAL)
 
-# This allows image specific sysbuild.cmake to be processed.
+# Check for any duplicate entries in image names to prevent an infinite loop
+set(TMP_IMAGES ${IMAGES})
+list(REMOVE_DUPLICATES TMP_IMAGES)
 list(LENGTH IMAGES images_length)
+list(LENGTH TMP_IMAGES tmp_images_length)
+
+if(NOT "${images_length}" EQUAL "${tmp_images_length}")
```

Comment:
This won't catch situations where an image adds an extra image with a colliding name here:
https://github.com/zephyrproject-rtos/zephyr/blob/3e98f20d66b8558e23fc63447259da0f66d12d59/share/sysbuild/CMakeLists.txt#L78